ding/policy/mbpolicy/dreamer.py

```diff
@@ -247,11 +248,17 @@ def _forward_collect(self, data: dict, world_model, envstep, reset=None, state=N
         action = action.detach()
 
         state = (latent, action)
+        assert isinstance(world_model.action_type, str), "please specify the action type"
+        if world_model.action_type == 'discrete':
+            action = torch.where(action == 1)[1]
         output = {"action": action, "logprob": logprob, "state": state}
 
         if self._cuda:
             output = to_device(output, 'cpu')
         output = default_decollate(output)
+        if world_model.action_type == 'discrete':
+            for l in range(len(output)):
+                output[l]['action'] = output[l]['action'].squeeze(0)
```

Comment:
can we employ squeeze operation before `default_decollate` operation? which may simplify this implementation